enterprise/core-edge/src/main/java/org/neo4j/coreedge/raft/replication/tx/ReplicatedTransactionStateMachine.java

```diff
@@ -84,21 +88,41 @@ public synchronized void flush() throws IOException
 
         if ( currentTokenId != txLockSessionId && txLockSessionId != Locks.Client.NO_LOCK_SESSION_ID )
         {
-            return Optional.of( Result.of( new TransactionFailureException(
-                    LockSessionExpired, "The lock session in the cluster has changed: [current lock session id:%d, tx lock session id:%d]",
+            callback.accept( Result.of( new TransactionFailureException( LockSessionExpired,
+                    "The lock session in the cluster has changed: [current lock session id:%d, tx lock session id:%d]",
                     currentTokenId, txLockSessionId ) ) );
         }
+        else
+        {
+            try
+            {
+                TransactionToApply transaction = new TransactionToApply( tx );
+                transaction.onClose( txId -> callback.accept( Result.of( txId ) ) );
```

Comment:
No. For the same reason you don't pass next in construction, it is not mandatory.